tests/system/test_blob.py

```diff
@@ -235,6 +235,52 @@ def test_blob_crud_w_user_project(
         blob1.delete()
 
 
+def test_blob_crud_w_etag_match(
+    shared_bucket, blobs_to_delete, file_data, service_account,
+):
+    wrong_etag = "kittens"
+
+    blob = shared_bucket.blob("SmallFile")
+
+    info = file_data["simple"]
+    with open(info["path"], mode="rb") as to_read:
+        payload = to_read.read()
+
+    blob.upload_from_filename(info["path"])
+    blobs_to_delete.append(blob)
+    etag = blob.etag
+
+    blob = shared_bucket.blob("SmallFile")
```

Comment:
Please use a different name for the blob used to populate the bucket and the one used for testing the APIs passing ETags.

Reply:
I'm not sure I understand what you're asking for. The blob on this line has to use the same name as the one at the top of the test. I just have to reset the blob to clear out the metadata. Well, I guess I don't _have_ to. And if I do I should probably do it before each exists/reload/download call. Or I could just remove this line altogether.

Reply:
Not a different blob name (`"SmallFile"`), but a different *variable* name, e.g. `blob` and `other_blob` or some such, to make it clear that we are starting from scratch.

Reply:
Ah, will do.